drivers/sensor/max17048/max17048.c

```diff
@@ -0,0 +1,188 @@
+LOG_MODULE_REGISTER(MAX17048, CONFIG_SENSOR_LOG_LEVEL);
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "MAX17048 driver enabled without any devices"
+#endif
+
+struct max17048_data {
+	float charge;
+	float voltage;
+	uint16_t time_to_full;
+	uint16_t time_to_empty;
+	bool charging;
+};
+
+/* Read buffer */
+unsigned char max17048_buffer[2];
+
+/**
+ * I2C communication
+ * The way we read a value is first writing the address we want to read and then
+ * wait for 2 bytes containing the data.
+ */
+uint16_t max17048_read_register(const struct device *dev, uint8_t registerId) {
+
+	const struct max17048_config *cfg = dev->config;
+
+	if (i2c_write_read_dt(&cfg->i2c, &registerId, sizeof(registerId), max17048_buffer, sizeof(max17048_buffer))) {
+		printk("Fail to fetch sample from sensor\n");
+		return -1;
+	}
+
+	return toUint16(max17048_buffer[0], max17048_buffer[1]);
+}
+
+/* Raw value from the internal ADC */
+uint16_t max17048_adc(const struct device *i2c_dev) {
+	return max17048_read_register(i2c_dev, REGISTER_VCELL);
+}
+
+/* Battery voltage */
+float max17048_voltage(const struct device *i2c_dev) {
+	return (float) max17048_adc(i2c_dev)  * 78.125 / 1000000;
+}
+
+/* Battery percentaje */
+float max17048_percent(const struct device *i2c_dev) {
+	float register_value = (float)max17048_read_register(i2c_dev, REGISTER_SOC);
+	return register_value / 256;
+}
+
+/* % per hour, positive is charging or negative if discharging */
+float max17048_crate(const struct device *i2c_dev) {
+	int16_t value = (int16_t) max17048_read_register(i2c_dev, REGISTER_CRATE);
+	return (float)value * 0.208f;
+}
+
+static int max17048_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct max17048_data *data = dev->data;
+	data->charge = max17048_percent(dev);
+	data->voltage = max17048_voltage(dev);
+
+	/* Crate is the current percentage of the battery charged or drained per hour */
+	float crate = max17048_crate(dev);
+	data->charging = crate > 0;
+
+	if (data->charging) {
+		uint8_t percentage_pending = 100 - data->charge;
+		float hours_pending = percentage_pending / crate;
+		data->time_to_empty = 0;
+		data->time_to_full = hours_pending * 60;
+	} else {
+		/* Discharging */
+		float hours_pending = data->charge / -crate;
+		data->time_to_empty = hours_pending * 60;
+		data->time_to_full = 0;
+	}
+
+	return 0;
+}
+
+static int max17048_channel_get(const struct device *dev, enum sensor_channel chan, struct sensor_value *val) {
+
+	uint8_t whole_part;
+	int decimal_part;
+
+	struct max17048_data *data = dev->data;
+	switch (chan) {
+		case SENSOR_CHAN_GAUGE_STATE_OF_CHARGE:
+			whole_part = data->charge;
+			decimal_part = (data->charge - whole_part) * 100 * 10000;
+
+			val->val1 = whole_part;
+			val->val2 = decimal_part;
+		break;
+
+		case SENSOR_CHAN_GAUGE_VOLTAGE:
+			whole_part = data->voltage;
+			decimal_part = (data->voltage - whole_part) * 100 * 10000;
+
+			val->val1 = whole_part;
+			val->val2 = decimal_part;
+		break;
```

Comment:
The formatting for the break statements looks a little weird. Could you run a `clang-format`?